rclcpp_examples/test/test_single_executable_example.py.in

```diff
@@ -0,0 +1,75 @@
+import io
+import os
+from launch import LaunchDescriptor
+from launch.exit_handler import primary_exit_handler
+from launch.launcher import DefaultLauncher
+from launch.output_handler import FileOutput
+from launch.output_handler import LineOutput
+
+
+class InMemoryHandler(LineOutput):
+    def __init__(self):
+        super(LineOutput, self).__init__()
+        self.left_over_stdout = b''
+        self.left_over_stderr = b''
+        self.stdout_data = io.BytesIO()
+        self.stderr_data = io.BytesIO()
+
+    def on_stdout_lines(self, lines):
+        self.stdout_data.write(lines)
+
+    def on_stderr_lines(self, lines):
+        self.stderr_data.write(lines)
+
+    def get_description(self):
+        return 'InMemoryHandler'
+
+
+def setup():
+    os.environ['OSPL_VERBOSITY'] = '8'  # 8 = OS_NONE
+
+
+def test_executable():
+    output_handler = InMemoryHandler()
+
+    ld = LaunchDescriptor()
+
+    ld.add_process(
+        cmd=['@RCLCPP_EXAMPLES_EXECUTABLE@', 'test_executable'],
+        name='test_executable',
+        exit_handler=primary_exit_handler,
```

Comment:
For a single process the `primary_exit_handler` doesn't make any difference. But when adding a second process they can't both be primary. Therefore I would recommend to use the default exit handler instead.